diff --git a/CMakeLists.txt b/CMakeLists.txt
index ff6424b..1234567 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1161,8 +1161,11 @@
     check_symbol_exists(fdatasync "unistd.h" HAVE_FDATASYNC)
     check_symbol_exists(gethostname "unistd.h" HAVE_GETHOSTNAME)
     check_symbol_exists(getpagesize "unistd.h" HAVE_GETPAGESIZE)
+    cmake_push_check_state(RESET)
+    list(APPEND CMAKE_REQUIRED_DEFINITIONS -D_GNU_SOURCE)
     check_symbol_exists(getresgid "unistd.h" HAVE_GETRESGID)
     check_symbol_exists(getresuid "unistd.h" HAVE_GETRESUID)
+    cmake_pop_check_state()
     check_symbol_exists(sigaction "signal.h" HAVE_SIGACTION)
     check_symbol_exists(sigtimedwait "signal.h" HAVE_SIGTIMEDWAIT)
     check_symbol_exists(setjmp "setjmp.h" HAVE_SETJMP)
